### ISSUE TITLE
Building system on branch master fails, while 2.4 branch works

### ISSUE THREAD

**Issue by andreasBihlmaier** [Itseez/opencv_contrib#105]
After trying to debug this for several hours, I ask for your help:

The cvv module builds fine against the opencv 2.4 branch.
After updating cvv for opencv master branch (see cvv_opencv_3.0.0 branch https://github.com/andreasBihlmaier/opencv_contrib/tree/cvv_opencv_3.0.0),
it compiles just fine against current master with an external CMakeLists.txt (https://github.com/CVVisualPSETeam/CVVisual/blob/master/CMakeLists.txt).
However, building it as extra module results in a totally broken libopencv_cvv.so [1], which manifests itself as a linker error for the cvv example [2]. I hope the diff shows why I believe this to be a problem in the opencv build system and not in the cvv module (the compare version works with the 2.4 branch): https://github.com/andreasBihlmaier/opencv_contrib/compare/andreasBihlmaier:cvv_fix_docs...cvv_opencv_3.0.0

[1]:

```
> objdump -tT lib/libopencv_cvv.so | grep showImage
0000000000046ea0 l     F .text  000000000000000c              _ZN3cvv4impl9showImageERKN2cv11_InputArrayERKNS0_12CallMetaDataEPKcS9_
> objdump -tT lib/libopencv_cvv.so | grep showImage | c++filt
0000000000046ea0 l     F .text  000000000000000c              cvv::impl::showImage(cv::_InputArray const&, cvv::impl::CallMetaData const&, char const*, char const*)

> readelf -Ws lib/libopencv_cvv.so | grep showImage
2642: 0000000000046ea0    12 FUNC    LOCAL  DEFAULT   11 _ZN3cvv4impl9showImageERKN2cv11_InputArrayERKNS0_12CallMetaDataEPKcS9_
> readelf -Ws lib/libopencv_cvv.so | grep showImage | c++filt
2642: 0000000000046ea0    12 FUNC    LOCAL  DEFAULT   11 cvv::impl::showImage(cv::_InputArray const&, cvv::impl::CallMetaData const&, char const*, char const*)


> nm -D -C lib/libopencv_cvv.so | grep -i showimage
#nothing


# build as extra module with master branch
> nm -D -C lib/libopencv_cvv.so | wc -l
967

# build with external CMakeLists.txt
> nm -D -C libcvvisual.so | wc -l
18544
> nm -D -C libcvvisual.so  | grep -i showimage
00000000002e4efe T cvv::gui::OverviewTable::showImages()
00000000003a309b T cvv::impl::showImage(cv::_InputArray const&, cvv::impl::CallMetaData const&, char const*, char const*)
```

[2]:

```
Linking CXX executable ../../bin/example_cvv_cvv_demo
cd /home/ahb/software/opencv/build/modules/cvv && /usr/bin/cmake -E cmake_link_script CMakeFiles/example_cvv_cvv_demo.dir/link.txt --verbose=true
/usr/bin/c++   -std=c++11   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wundef -Winit-self -Wpointer-arith  -Wsign-promo -Wno-narrowing -Wno-delete-non-virtual-dtor -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -msse -msse2 -mavx -ffunction-sections -fvisibility=hidden -fvisibility-inlines-hidden -std=c++11 -Wall -Wextra -pedantic -Wno-shadow -O2 -g -DNDEBUG     CMakeFiles/example_cvv_cvv_demo.dir/samples/cvv_demo.cpp.o CMakeFiles/example_cvv_cvv_demo.dir/example_cvv_cvv_demo_automoc.cpp.o  -o ../../bin/example_cvv_cvv_demo -rdynamic ../../lib/libopencv_cvv.so.3.0.0 ../../lib/libopencv_core.so.3.0.0 ../../lib/libopencv_flann.so.3.0.0 ../../lib/libopencv_imgproc.so.3.0.0 ../../lib/libopencv_imgcodecs.so.3.0.0 ../../lib/libopencv_videoio.so.3.0.0 ../../lib/libopencv_highgui.so.3.0.0 ../../lib/libopencv_ml.so.3.0.0 ../../lib/libopencv_features2d.so.3.0.0 ../../lib/libopencv_imgcodecs.so.3.0.0 ../../lib/libopencv_videoio.so.3.0.0 ../../lib/libopencv_highgui.so.3.0.0 ../../lib/libopencv_flann.so.3.0.0 ../../lib/libopencv_videoio.so.3.0.0 ../../lib/libopencv_imgcodecs.so.3.0.0 ../../lib/libopencv_imgproc.so.3.0.0 ../../lib/libopencv_ml.so.3.0.0 ../../lib/libopencv_core.so.3.0.0 -Wl,-rpath,/home/ahb/software/opencv/build/lib 
CMakeFiles/example_cvv_cvv_demo.dir/samples/cvv_demo.cpp.o: In function `debugDMatch':
/home/ahb/software/opencv_contrib/modules/cvv/include/opencv2/cvv/dmatch.hpp:36: undefined reference to `cvv::impl::debugDMatch(cv::_InputArray const&, std::vector<cv::KeyPoint, std::allocator<cv::KeyPoint> >, cv::_InputArray const&, std::vector<cv::KeyPoint, std::allocator<cv::KeyPoint> >, std::vector<cv::DMatch, std::allocator<cv::DMatch> >, cvv::impl::CallMetaData const&, char const*, char const*, bool)'
CMakeFiles/example_cvv_cvv_demo.dir/samples/cvv_demo.cpp.o: In function `showImage':
/home/ahb/software/opencv_contrib/modules/cvv/include/opencv2/cvv/show_image.hpp:29: undefined reference to `cvv::impl::showImage(cv::_InputArray const&, cvv::impl::CallMetaData const&, char const*, char const*)'
CMakeFiles/example_cvv_cvv_demo.dir/samples/cvv_demo.cpp.o: In function `debugFilter':
/home/ahb/software/opencv_contrib/modules/cvv/include/opencv2/cvv/filter.hpp:31: undefined reference to `cvv::impl::debugFilter(cv::_InputArray const&, cv::_InputArray const&, cvv::impl::CallMetaData const&, char const*, char const*)'
CMakeFiles/example_cvv_cvv_demo.dir/samples/cvv_demo.cpp.o: In function `finalShow':
/home/ahb/software/opencv_contrib/modules/cvv/include/opencv2/cvv/final_show.hpp:31: undefined reference to `cvv::impl::finalShow()'
collect2: error: ld returned 1 exit status
```

**Comment by StevenPuttemans** [comment]
Since the ccv module is actually not even an option in cmake configuration, should this still be open? The module is there but there is no way of building it?

**Comment by andreasBihlmaier** [comment]
@StevenPuttemans the cvv module is there and thanks to @mshabunin (thank you *very* much!) compiles just fine against current opencv master branch.

This issue can thus be closed :)

**Comment by StevenPuttemans** [comment]
:+1:

**Comment by mshabunin** [comment]
Should be fixed in #163
